### PR TITLE
fix: spark.executor.cores' default value based on master when counting workers

### PR DESCRIPTION
**Pull request by ocworld**
This pr is fixing spark.executor.cores' default value based on master when counting workers.

The default value of spark.executor.cores is different as what master is.

It is 1 in YARN mode. In other case, It is all the available cores on the worker in standalone and Mesos coarse-grained modes.
(https://spark.apache.org/docs/latest/configuration.html)

Currently, I've tried to use mmlspark based on spark on k8s. When I did not set spark.executor.cores, error was occured. It is because not matched expected num of mmlspark workers with actual value.

### CODE REVIEW

**Comment by imatiach-msft** [comment]
will need to test this out a bit since there have been a lot of issues in the past around this specific code section

**Comment by imatiach-msft** [comment]
the build was all green but it's not updating in github, weird, looks like some github bug

**Comment by ocworld** [comment]
@imatiach-msft I wrote new unittest and pushed. However, It is hard to tested mmlspark's unittest in my local environment . Please check it :)

**Comment by ocworld** [comment]
@imatiach-msft Can I see why the build failed?

**Comment by imatiach-msft** [comment]
@ocworld the style checker is failing on:

[error] /home/vsts/work/1/s/src/test/scala/com/microsoft/ml/spark/core/utils/VerifyClusterUtil.scala: Header does not match expected text

It looks like you need to add this copyright message at the top of the file to pass the style checker:
```
// Copyright (C) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See LICENSE in project root for information.
```

File: src/main/scala/com/microsoft/ml/spark/core/utils/ClusterUtil.scala [hunk 1]
```diff
@@ -19,37 +19,91 @@ object ClusterUtil {
...
+        val coresPerExec = defaultNumCores / confTaskCpus
         log.info(s"ClusterUtils calculated num cores per executor as $coresPerExec from " +
-          s"$numMachineCores machine cores from JVM and $confTaskCpus task CPUs")
+          s"default num cores($defaultNumCores) from master and $confTaskCpus task CPUs")
         coresPerExec
     }
   }
 
+  /** Get number of default cores from sparkSession(required) or master(optional) for 1 executor.
+    * @param spark The current spark session. If not set master param, the master in the session is used.
```
Comment (imatiach-msft) [inline src/main/scala/com/microsoft/ml/spark/core/utils/ClusterUtil.scala:41]: nit: If not set master param, the master in the session is used. ->
If master parameter is not set, the master in the spark session is used.

**Comment by imatiach-msft** [comment]
@ocworld I tried this PR out on a cluster (azure databricks) with Higgs dataset and it didn't change the training time for me (compared to previous runs), so it seems safe to me at least in that scenario.  I think they use spark standalone mode for databricks clusters.

File: src/main/scala/com/microsoft/ml/spark/core/utils/ClusterUtil.scala (hunk 1 above)
Comment (ocworld) [reply src/main/scala/com/microsoft/ml/spark/core/utils/ClusterUtil.scala]: It is fixed

**Comment by ocworld** [comment]
> @ocworld the style checker is failing on:
> 
> [error] /home/vsts/work/1/s/src/test/scala/com/microsoft/ml/spark/core/utils/VerifyClusterUtil.scala: Header does not match expected text
> 
> It looks like you need to add this copyright message at the top of the file VerifyClusterUtil.scala to pass the style checker:
> 
> ```
> // Copyright (C) Microsoft Corporation. All rights reserved.
> // Licensed under the MIT License. See LICENSE in project root for information.
> ```

Copyright is added in the file

**Comment by ocworld** [comment]
> @ocworld I tried this PR out on a cluster (azure databricks) with Higgs dataset and it didn't change the training time for me (compared to previous runs), so it seems safe to me at least in that scenario. I think they use spark standalone mode for databricks clusters.

Thanks for your test

**Comment by imatiach-msft** [comment]
it looks like @mhamilton723 is required to review as a code owner (can't seem to merge), can you please review this PR @mhamilton723 ?

**Comment by ocworld** [comment]
@ahnlab-oss

**Pull request merged**